Numbers/ViewController/GridViewController.swift

```diff
@@ -0,0 +1,296 @@
+    @IBOutlet var tapGesture1: UITapGestureRecognizer!
+    
+    var boxList = [UILabel]()
+    var boxOrder = [Int]()
+    var level = 4
+    @IBOutlet weak var levelLabel: UILabel!
+    var tapCount = 0
+    var livesLeft = 2
+    let secs = 0.7
+    
+    // Screen width.
+    public var screenWidth: CGFloat {
+        return UIScreen.main.bounds.width
+    }
+    
+    // Screen height.
+    public var screenHeight: CGFloat {
+        return UIScreen.main.bounds.height
+    }
+    
+    
+    override func viewDidLoad() {
+        incorrectLabel.isHidden = true
+        super.viewDidLoad()
+        setGrid(color: UIColor.black.cgColor)
+        setArray()
+        setLevel()
+        tapGesture1.isEnabled = false
+        preStart()
+        start()
+
+    }
+    
+    func setGrid(color: CGColor) {
+        //vertical lines
+        addLine(fromPoint: CGPoint(x: screenWidth * (1/3), y: 0), toPoint: CGPoint(x: screenWidth * (1/3), y: screenHeight), color: color)
+        addLine(fromPoint: CGPoint(x: screenWidth * (2/3), y: 0), toPoint: CGPoint(x: screenWidth * (2/3), y: screenHeight), color: color)
+        
+        //horizontal lines
+        addLine(fromPoint: CGPoint(x: 0, y: screenHeight * (1/4)), toPoint: CGPoint(x: screenWidth, y: screenHeight * (1/4)), color: color)
+        addLine(fromPoint: CGPoint(x: 0, y: screenHeight * (2/4)), toPoint: CGPoint(x: screenWidth, y: screenHeight * (2/4)), color: color)
+        addLine(fromPoint: CGPoint(x: 0, y: screenHeight * (3/4)), toPoint: CGPoint(x: screenWidth, y: screenHeight * (3/4)), color: color)
+    }
+    
+    func addLine(fromPoint start: CGPoint, toPoint end:CGPoint, color: CGColor) {
+        let line = CAShapeLayer()
+        let linePath = UIBezierPath()
+        linePath.move(to: start)
+        linePath.addLine(to: end)
+        line.path = linePath.cgPath
+        line.strokeColor = color
+        line.lineWidth = 1
+        line.lineJoin = CAShapeLayerLineJoin.round
+        self.view.layer.addSublayer(line)
+    }
+    
+    func setArray() {
+        boxList.append(box1)
+        boxList.append(box2)
+        boxList.append(box3)
+        boxList.append(box4)
+        boxList.append(box5)
+        boxList.append(box6)
+        boxList.append(box7)
+        boxList.append(box8)
+        boxList.append(box9)
+        boxList.append(box10)
+        boxList.append(box11)
+        boxList.append(box12)
+    }
+    
+    func setLevel() {
+      levelLabel.text = "Level " + String(level - 3)
+    }
+    
+    func preStart() {
+        for index in 0...2 {
+            var randomNum = Int.random(in: 0 ..< boxList.count)
+            if(boxOrder.count > 0) {
+                while(randomNum == boxOrder[boxOrder.count - 1]) {
+                    randomNum = Int.random(in: 0 ..< boxList.count)
+                }
+            }
+            boxOrder.append(randomNum)
+        }
+    }
+    
+    
+    func start() {
+        var randomNum = Int.random(in: 0 ..< boxList.count)
```

Comment:
This is very similar to the previous function, pull these out into a common function boxOrder.append(chooseRandomBox)

Reply:
Also, start() is not a good name for this.  runNextLevel() would be better